test/lib/TestMac.py

```diff
@@ -25,55 +25,70 @@ def CheckFileType(test, file, archs):
     pattern = re.compile('^Architectures in the fat file: (.*) are: (.*)$')
   match = pattern.match(o)
   if match is None:
-    print('Ouput does not match expected pattern: %s' % (pattern.pattern))
+    print('Output does not match expected pattern: %s' % pattern.pattern)
     test.fail_test()
   else:
     found_file, found_archs = match.groups()
     if found_file != file or set(found_archs.split()) != set(archs):
-      print('Expected file %s with arch %s, got %s with arch %s' % (
-          file, ' '.join(archs), found_file, found_archs))
+      print('Expected file %s with arch %s, got %s with arch %s' % (file, ' '.join(archs), found_file, found_archs))
       test.fail_test()
 
 
-class XcodeInfo(object):
-  """Simplify access to Xcode informations."""
+def run(*cmd_args):
+  return subprocess.check_output(cmd_args, stderr=subprocess.PIPE).decode('utf-8')
+
+
+class Xcode(object):
+  """Simplify access to Xcode information."""
+  _cache = {}
+
+  @staticmethod
+  def Version():
+    if 'Version' not in Xcode._cache:
+      version = ''
+      try:
+        lines = run('xcodebuild', '-version').splitlines()
+        version = ''.join(lines[0].decode('utf-8').split()[-1].split('.'))
+        version = (version + '0' * (3 - len(version))).zfill(4)
+      except subprocess.CalledProcessError:
+        pass
+      try:
+        lines = run('pkgutil', '--pkg-info=com.apple.pkg.CLTools_Executables').splitlines()
+        for l in lines:
+          n, v = l.split(': ', 1)
+          if n != 'version':
+            continue
+          parts = v.split('.',4)
+          version = '%s%s%s%s' % tuple(parts[0:4])
+          break
+      except subprocess.CalledProcessError:
+        pass
+      Xcode._cache['Version'] = version
+    return Xcode._cache['Version']
```

Comment:
```
    return version
return Xcode._cache.get('Version', Version())
```

Reply:
I was thinking about that but I wanted to keep the lazy evaluate of the "else"...
Actually this is a classic place for `@memoize`...